### ISSUE TITLE
Menu Component

### ISSUE THREAD

**Issue by arsh-cedcoss** [bigcommerce/big-design#419]
Required a Menu component for listing the different sections which allow users to move around the site quickly and efficiently.
Inside the Menu (Parent) Component, there should be MenuItem to list the sections and also Submenu for listing the childs of MenuItem. The Component should have responsive. and accept the props for alignment.
Horizontal for Navbar and Vertical for Sidebar.

**Comment by chanceaclark** [comment]
Hey @arsh-cedcoss, thanks for bringing this up. I'll defer this to our designers to see if there is pattern we might have in the future. In the meantime, could you maybe provide a mockup/sketch/codesandbox with an example Menu component?
cc. @valterfatia @eugene-polev

**Comment by valterfatia** [comment]
Thanks for the ping @chanceaclark!

@arsh-cedcoss we're currently in the process of reviewing our navigation globally (including apps) so we haven't planned to provide this pattern in the short term, yet we also understand that the needs of our partners work at a different place… meaning we may provide this pattern but it would be a temporary solution. 

As Chance mentioned it would be good to understand what you're trying to achieve so we can tailor our guidance.

Looking into the future our plan is to streamline navigation into 3 components:
- Global sidebar, including extension points for apps
- Vertical "index", to allow quick navigation on a long page
- Horizontal tabs, to separate differentiated content into different views

**Comment by arsh-cedcoss** [comment]
Thanks, @chanceaclark & @valterfatia for giving attention to this issue. For the time being, I'm using the custom navbar which is third-party code. I want this **(attached image)** kind of navbar with responsive behavior and it should have a prop so that it could be a vertical side-bar too. I hope your future navigation components will help me to achieve this.

![1597340194245](https://user-images.githubusercontent.com/53248675/90171622-628b2680-ddbf-11ea-8445-ed4b07517392.JPEG)
![1597340194352](https://user-images.githubusercontent.com/53248675/90171628-6454ea00-ddbf-11ea-9bc4-9c7f0a2f644f.JPEG)
![Screenshot (46)](https://user-images.githubusercontent.com/53248675/90171631-64ed8080-ddbf-11ea-95cb-82e5bef8c073.png)

**Comment by valterfatia** [comment]
Tracked internally on BIGDESIGN-214

**Comment by arsh-cedcoss** [comment]
@valterfatia is it on development stage?

**Comment by valterfatia** [comment]
@arsh-cedcoss not at the moment, but we added it to a wider topic on navigation since we are looking into way to provide a more integrated experience for apps to build their navigation inside control panel. Out of curiosity did you have a specific use case for this component?

**Comment by arsh-cedcoss** [comment]
Hey @valterfatia
 As there are some sections in the app at the parent level which need Menu Component for navigation. The component behavior should be compatible with to control panel & mobile and have a property of alignment vertical (Sidebar) & horizontal (Nav Bar).

**Comment by valterfatia** [comment]
Got it! This type of navigation will be enabled by the BigCommerce sidebar which we are working on making extensible by apps. Some of this functionality is already available but exclusively within Channels.

**Comment by arsh-cedcoss** [comment]
Yeah, I've seen it in the 'Channel Manager' Section, Hope it will release soon!
